src/main.cpp

```diff
@@ -412,138 +526,106 @@ GbdxmArgs* readPackArgs(const po::variables_map& vm)
         }
-        exit(1);
+    } else if(categories.size() > 1) {
+        errors.push_back("Please specify a model category, possible categories "
+                         "for this model are: " + join(categories, ", "));
+    } else {
+        metadata.setCategory(categories[0]);
     }
 
+    DG_CHECK(errors.empty(), "%s", join(errors, "\n").c_str());
+
     if(vm.count("plaintext")) {
         args->encrypt = false;
     }
 
-    return args.release();
+    return args;
 }
 
 vector<string> readJsonMetadata(const string& fileName, GbdxmPackArgs& args)
 {
-    if(!exists(fileName) || is_directory(fileName)) {
-        cerr << "Json file does not exist at '" << fileName << "'" << endl;
-        exit(1);
-    }
+    DG_CHECK(exists(fileName), "%s does not exist");
+    DG_CHECK(!is_directory(fileName), "%s is a directory");
 
     vector<string> missingFields;
-    try {
-        ifstream ifs(fileName);
-        Reader reader;
-        Value root;
-        if(!reader.parse(ifs, root)) {
-            cerr << "Error parsing metadata: " << reader.getFormattedErrorMessages() << endl;
-            exit(1);
-        }
+    ifstream ifs(fileName);
+    DG_CHECK(ifs.good(), "Error opening %s: %s", fileName.c_str(), strerror(errno));
 
-        args.metadata = ModelMetadataJson::fromJsonPartial(root, missingFields);
+    Json::Reader reader;
+    Json::Value root;
+    DG_CHECK(reader.parse(ifs, root), "Error parsing metadata: %s",
+             reader.getFormattedErrorMessages().c_str());
 
-        if(root.isMember("content")) {
-            if(root["content"].type() != Json::objectValue) {
-                cerr << "Invalid metadata content field: must be a JSON object." << endl;
-            }
+    auto metadata = classification::ModelMetadataJson::fromJsonPartial(root, missingFields);
+    args.package = classification::ModelPackage::create(move(metadata));
 
-            for(const auto& name : root["content"].getMemberNames()) {
-                args.modelFiles[name] = root["content"][name].asString();
-            }
-        }
+    if(root.isMember("content")) {
+        DG_CHECK(root["content"].type() == Json::objectValue,
+                 "Invalid metadata \"content\" field: must be a JSON object.");
 
-    } catch(const std::exception& e) {
-        cerr << "Error reading metadata from " << fileName << ": " << e.what() << endl;
-        exit(1);
+        for(const auto& name : root["content"].getMemberNames()) {
+            args.modelFiles[name] = root["content"][name].asString();
+        }
     }
 
     return missingFields;
 }
 
 void readModelMetadata(GbdxmPackArgs& args, vector<string>& missingFields)
 {
-    unique_ptr<ModelPackage> model(ModelPackage::create(*args.metadata));
-
-    // go through model files that can contain metadata
-    for(const auto& itemName : model->modelMetadataItemNames()) {
+    // Load the files with metadata into the ModelPackage
+    for(const auto& itemName : args.identifier->metadataItems()) {
```

Comment:
`metadataItems()` are items that contain metadata that can be extracted. In case of Caffe, we only extract metadata from the `.prototxt` file AKA the model.